### PR TITLE
added tests for descision_maker

### PR DESCRIPTION
**Pull request by Totoual**
## Proposed changes

Adds test in order to achieve 100% coverage on decision_maker.

## Fixes

Part of #514 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_connections/test_oef/test_communication.py [hunk 1]
```diff
@@ -317,6 +317,7 @@ def setup_class(cls):
 
     def test_cfp(self):
         """Test that a CFP can be sent correctly."""
+        logger.info("Inside the cfp function!")
```
Comment (marcofavorito) [inline tests/test_connections/test_oef/test_communication.py]: I think this is a debugging message, right? Can it be removed?

File: tests/test_decision_maker/test_base.py [hunk 2]
```diff
@@ -66,6 +67,23 @@ def test_initialisation(self):
...
 
+    def test_body(self):
+        """Test the setter for the body."""
+        msg = InternalMessage()
+        msg.body = {"test_key": "test_value"}
+
+        other_msg = InternalMessage(body={"test_key": "test_value"})
+        assert msg == other_msg, "Messages should be equal."
+        assert msg.check_consistency(), "It is true."
+        assert str(msg) == msg.__str__()
```
Comment (marcofavorito) [inline tests/test_decision_maker/test_base.py]: `str(obj)` does call `msg.__str__()`. So the string function is not being really tested.

Ideally, we should have something like:
```python
assert str(msg) == "the string representation of the message that you expect"
```

File: tests/test_decision_maker/test_base.py [hunk 3]
```diff
@@ -396,6 +414,61 @@ def test_decision_maker_execute_w_wrong_input(self):
...
+                                        counterparty_tx_fee=0,
+                                        ledger_id="off_chain",
+                                        quantities_by_good_id={"good_id": 10},
+                                        info={'some_info_key': 'some_info_value'})
+
+        tx_digest = self.decision_maker._settle_tx(tx_message)
+        assert tx_digest == tx_message.transaction_id
+
+    def test__is_utility_enhancing(self):
+        """Test the off_chain message."""
```
Comment (marcofavorito) [inline tests/test_decision_maker/test_base.py]: fix docstring

File: tests/test_crypto/test_ethereum_crypto.py [hunk 4]
```diff
@@ -45,5 +46,6 @@ def test_initialization():
 def test_sign_transaction():
     """Test the signing function for the eth_crypto."""
     account = EthereumCrypto(PRIVATE_KEY_PATH)
-    sign_bytes = account.sign_transaction(b'Hello')
+    tx = Web3.solidityKeccak(['bytes'], [b'hello'])
+    sign_bytes = account.sign_transaction(tx)
```
Comment (Totoual) [inline tests/test_crypto/test_ethereum_crypto.py]: Kind of "hacky" but it works.

File: tests/test_decision_maker/test_base.py [hunk 5]
```diff
@@ -404,55 +404,49 @@ def test_decision_maker_execute_w_wrong_input(self):
...
-                                        currency_id="FET",
-                                        amount=2,
-                                        sender_tx_fee=0,
-                                        counterparty_tx_fee=0,
+        tx_message = TransactionMessage(performative=TransactionMessage.Performative.PROPOSE_FOR_SETTLEMENT,
+                                        skill_callback_ids=["default"],
+                                        tx_id="transaction0",
+                                        tx_sender_addr="agent_1",
+                                        tx_counterparty_addr="pk",
+                                        tx_amount_by_currency_id={"good_id": 10},
```
Comment (DavidMinarsch) [inline tests/test_decision_maker/test_base.py]: needs to be {`FET`: -20} or something like that

File: tests/test_decision_maker/test_base.py [hunk 6]
```diff
@@ -404,55 +404,49 @@ def test_decision_maker_execute_w_wrong_input(self):
...
```
Comment (DavidMinarsch) [inline tests/test_decision_maker/test_base.py]: see above

File: tests/test_decision_maker/test_base.py [hunk 7]
```diff
@@ -404,55 +404,49 @@ def test_decision_maker_execute_w_wrong_input(self):
...
-                                        amount=2,
-                                        sender_tx_fee=0,
-                                        counterparty_tx_fee=0,
+        """Test the utility enhancing for off_chain message."""
+        tx_message = TransactionMessage(performative=TransactionMessage.Performative.PROPOSE_FOR_SETTLEMENT,
+                                        skill_callback_ids=["default"],
+                                        tx_id="transaction0",
+                                        tx_sender_addr="agent_1",
+                                        tx_counterparty_addr="pk",
+                                        tx_amount_by_currency_id={"good_id": 10},
```
Comment (DavidMinarsch) [inline tests/test_decision_maker/test_base.py]: see above

**Pull request merged**